### PR TITLE
[django] Generalize uuid/key logic openwisp/openwisp-utils#40

### PR DESCRIPTION
**Pull request by waleko**
Removed the logics which were duplicated
in the others modules and redefined it here to enable reusability and ease debugging.

Related to openwisp/openwisp-utils#40

### CODE REVIEW

**Comment by waleko** [comment]
This commit also has a lot of `isort` fixes

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,9 @@ before_install:
 
 install:
   - pip install $DJANGO
+  # Temporary: remove when openwisp-utils#71 is merged
+  - pip install "https://github.com/waleko/openwisp-utils/tarball/issues/40" --upgrade
```
Comment (nemesisdesign) [inline .travis.yml]: I released openwisp-utils 0.4.1 which contains your patch https://github.com/openwisp/openwisp-utils/pull/71.
You can now remove this temporary addition.

File: requirements-test.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 coveralls
-openwisp-utils[qa]>=0.3.1
+openwisp-utils[qa]>=0.4.0
```
Comment (nemesisdesign) [inline requirements-test.txt]: use 0.4.1

File: requirements.txt [hunk 3]
```diff
@@ -7,4 +7,4 @@ django-sortedm2m==2.0.0
 django-reversion>=2.0.6,<3.0.5
 django-x509>=0.5.0,<0.6.0
 django-taggit>=0.23.0,<0.24.0
-openwisp-utils>=0.2,<0.3
+openwisp-utils>=0.4,<0.5
```
Comment (nemesisdesign) [inline requirements.txt]: use 0.4.1

**Pull request merged**